pkg/connector/helpers.go

```diff
@@ -75,8 +75,8 @@ type RateLimitInfo struct {
 	Remaining int64
 }
 
-func NewRateLimitInfo(limit, remaining int64) RateLimitInfo {
-	return RateLimitInfo{
+func NewRateLimitInfo(limit, remaining int64) *RateLimitInfo {
+	return &RateLimitInfo{
```

Comment:
Why is this change needed? The code looks uglier so it must be important. :)

Reply:
@ggreer it is not needed, I just like to use pointers as a good practice 😄 I'm not sure about using it with such small structs though. I reverted this change and use copies of `RateLimitInfo` struct. I could refactor it to use array of pointers, but that might be overkill, what do you think of using pointers over copies?